homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,184 @@
+                with Remote(config.copy()):
+                    LOGGER.debug("Working config: %s", config)
+                    self._method = method
+                    return RESULT_SUCCESS
+            except AccessDenied:
+                LOGGER.debug("Working but denied config: %s", config)
+                return RESULT_AUTH_MISSING
+            except UnhandledResponse:
+                LOGGER.debug("Working but unsupported config: %s", config)
+                return RESULT_NOT_SUPPORTED
+            except (OSError):
+                LOGGER.debug("Failing config: %s", config)
+
+        LOGGER.debug("No working config found")
+        return RESULT_NOT_FOUND
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        self._on_script = user_input.get(CONF_ON_ACTION)
+        self._port = user_input.get(CONF_PORT)
+
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            ip_address = await self.hass.async_add_executor_job(
+                _get_ip, user_input[CONF_HOST]
+            )
+
+            await self.async_set_unique_id(ip_address)
+            self._abort_if_unique_id_configured()
+
+            self._host = user_input.get(CONF_HOST)
+            self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+            self._title = user_input.get(CONF_NAME)
+
+            result = await self.hass.async_add_executor_job(self._try_connect)
+
+            if result != RESULT_SUCCESS:
+                return self.async_abort(reason=result)
+            return self._get_entry()
+
+        return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+    async def async_step_ssdp(self, user_input=None):
+        """Handle a flow initialized by discovery."""
+        host = urlparse(user_input[ATTR_SSDP_LOCATION]).hostname
+        ip_address = await self.hass.async_add_executor_job(_get_ip, host)
+
+        self._host = host
+        self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+        self._manufacturer = user_input[ATTR_UPNP_MANUFACTURER]
+        self._model = user_input[ATTR_UPNP_MODEL_NAME]
+        self._name = user_input[ATTR_UPNP_FRIENDLY_NAME]
+        if self._name.startswith("[TV]"):
+            self._name = self._name[4:]
+        self._title = f"{self._name} ({self._model})"
+        self._uuid = user_input[ATTR_UPNP_UDN]
+        if self._uuid.startswith("uuid:"):
+            self._uuid = self._uuid[5:]
+
+        config_entry = await self.async_set_unique_id(ip_address)
+        if config_entry:
+            config_entry.data[CONF_MANUFACTURER] = self._manufacturer
+            config_entry.data[CONF_MODEL] = self._model
+            config_entry.data[CONF_ID] = self._uuid
+            self.hass.config_entries.async_update_entry(config_entry)
```

Comment:
@MartinHjelmare This appears first after next restart. Is there a way to make this instantly visible?

Reply:
It should propagate immediately to the entry but the file is saved after a delay of 15 secs I think.

Reply:
If we need to update something immediately based on the config entry, register a listener.

https://developers.home-assistant.io/docs/en/config_entries_options_flow_handler.html#signal-updates